### ISSUE TITLE
Question on roadmap

### ISSUE THREAD

**Issue by gsvgit** [isaksky/FsSqlDom#1]
Hello!

What is goal of this project?
Is it aimed only to TSQL? What about PL-SQL and other dialects?
Do you interested in SQL static analysis tool creation?

**Comment by isaksky** [comment]
Hi, good question.

This library was created kind of opportunistically, since I saw there would be a pretty low effort way to support all of TSQL by leveraging existing work. If I had to do it all (TSQL parsing) from scratch, I would probably not do the project, since it is the type of problem that can take years, and people start companies around selling solutions for.

So for each other dialect, if I was to do it, I'd have to weigh popularity, ease/viability of metaprogramming the syntax tree. For PL-SQL, it is very popular, and I see that there is a standalone parser available. So it is a contender, but I'll need to do a little more research to see how difficult it would be. For TSQL it was pretty easy, since there was already a .NET assembly that I could just do reflection on, but I don't think so for PL-SQL.

In general, the project goal is to provide:
* a nice API around SQL parsing
* script generation (80/20 principle for this, since TSQL syntax tree is huge. Supporting the whole type tree would be a huge effort, but supporting a subset that would take care of 99% of needs would be fairly easy)

For SQL static analysis, I'm interested, but it may make sense to do in a another project downstream from this one, because in order to do a good job, you'd need a connection to the database to get schemas, view definitions, etc, and this is just a library.
